hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientSmartInvocationServiceImpl.java

```diff
@@ -45,10 +45,9 @@ public void invokeOnPartitionOwner(ClientInvocation invocation, int partitionId)
         if (owner == null) {
             throw new IOException("Partition does not have owner. partitionId : " + partitionId);
         }
-        ClientConnection connection =
-                (ClientConnection) connectionManager.getOrConnect(owner, authenticator);
         invocation.getClientMessage().setPartitionId(partitionId);
-        send(invocation, connection);
+        Connection connection = getConnection(owner);
```

Comment:
I am mostly trying to use Connection interface to be able to change the underlying implementation easily. Since it is a change in process, there are some places uses ClientConnection and some places that uses Connection.